### PR TITLE
Feature/190 trading application

### PR DESCRIPTION
**Pull request by AVLozhechkin**
Notes:
App structure looks like this:
изображение
Making transaction logic looks like this:
MakeTransactionImage
Changing share type looks like this:
ChangeTypeImage
Registering logic looks like this:
Register
Data model looks like this:
изображение

### CODE REVIEW

File: HW6/Artem_Lozhechkin/TradingApp.Core/Logger.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿namespace TradingApp.Core
+{
+    using log4net;
+    using log4net.Config;
+
+    public static class Logger
```
Comment (AVLozhechkin) [inline HW6/Artem_Lozhechkin/TradingApp.Core/Logger.cs]: Im not sure about Logger placement. Where it should be placed: in Core namespace or in TradingManager?

File: HW6/Artem_Lozhechkin/TradingApp.Core/LoggingServices/LoggingShareService.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
+﻿namespace TradingApp.Core.LoggingServices
+{
+    using System;
+    using System.Collections.Generic;
+    using TradingApp.Core.Models;
+    using TradingApp.Core.Repositories;
+    using TradingApp.Core.Services;
+
+    public class LoggingShareService : ShareService
```
Comment (AVLozhechkin) [inline HW6/Artem_Lozhechkin/TradingApp.Core/LoggingServices/LoggingShareService.cs]: Inheritance is used to log services with an access to the repositories.

File: HW6/Artem_Lozhechkin/TradingApp.Core/Repositories/IRepository.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    using System.Linq.Expressions;
+
+    public interface IRepository<T>
+    {
+        int Add(T entity);
+        int AddRange(IList<T> entities);
+        int Save(T entity);
+        int Delete(T entity);
+        T GetById(int? id);
+        //IEnumerable<T> GetByPredicate(Expression<Func<T, bool>> predicate);
```
Comment (AVLozhechkin) [inline HW6/Artem_Lozhechkin/TradingApp.Core/Repositories/IRepository.cs]: Is using methods like this (with predicates) a bad practice? It is easy to use, but I haven't found out how to test it,

**Comment by AVLozhechkin** [comment]
Also I have a question about initializer: App won't make transaction every 10 seconds (6 button in menu) if you will start with it.
изображение
I think it happens because SeedInitializer doesn't fill the database if you choose 6 as a first option. But I'm not sure why it doesnt fill.

File: HW6/Artem_Lozhechkin/TradingApp.Core/Logger.cs (hunk 1 above)
Comment (chesdenis) [reply HW6/Artem_Lozhechkin/TradingApp.Core/Logger.cs]: abstract logic should be in core. implementation logic should be in console or another application

File: HW6/Artem_Lozhechkin/TradingApp.Core/LoggingServices/LoggingShareService.cs (hunk 2 above)
Comment (chesdenis) [reply HW6/Artem_Lozhechkin/TradingApp.Core/LoggingServices/LoggingShareService.cs]: it is interesting thing which you are trying to replicate.  Have you find suitable pattern for that?

File: HW6/Artem_Lozhechkin/TradingApp.Core/Repositories/IRepository.cs (hunk 3 above)
Comment (chesdenis) [reply HW6/Artem_Lozhechkin/TradingApp.Core/Repositories/IRepository.cs]: expressions are not a part of this course, but of course your idea is really good example. can you please ask to describe more about it after my vacation? probably we have to try expressions for that.

**Pull request merged**